From 959cc1b85182864297b1526f8fac6906c2ccebd5 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Fri, 17 Aug 2018 11:00:59 +0800
Subject: [PATCH 4526/5242] MLK-19413-15 gpu: imx: dpu: extdst: Add
 extdst_is_master() helper support

commit  df63dadb239be99d63aa79b6cf170c93d132578a from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch adds extdst_is_master() helper support so that
callers may know if a extdst is a master extdst or not.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-extdst.c |    6 ++++++
 include/video/dpu.h              |    1 +
 2 files changed, 7 insertions(+)

diff --git a/drivers/gpu/imx/dpu/dpu-extdst.c b/drivers/gpu/imx/dpu/dpu-extdst.c
index d69ab5f..6c6555b 100644
--- a/drivers/gpu/imx/dpu/dpu-extdst.c
+++ b/drivers/gpu/imx/dpu/dpu-extdst.c
@@ -401,6 +401,12 @@ u32 extdst_perfresult(struct dpu_extdst *ed)
 }
 EXPORT_SYMBOL_GPL(extdst_perfresult);
 
+bool extdst_is_master(struct dpu_extdst *ed)
+{
+	return ed->id == 0;
+}
+EXPORT_SYMBOL_GPL(extdst_is_master);
+
 struct dpu_extdst *dpu_ed_get(struct dpu_soc *dpu, int id)
 {
 	struct dpu_extdst *ed;
diff --git a/include/video/dpu.h b/include/video/dpu.h
index c9bd78f..40d796c 100644
--- a/include/video/dpu.h
+++ b/include/video/dpu.h
@@ -546,6 +546,7 @@ void constframe_constantcolor(struct dpu_constframe *cf, unsigned int r,
 void extdst_pixel_cnt(struct dpu_extdst *ed, u16 *x, u16 *y);
 void extdst_last_pixel_cnt(struct dpu_extdst *ed, u16 *x, u16 *y);
 u32 extdst_perfresult(struct dpu_extdst *ed);
+bool extdst_is_master(struct dpu_extdst *ed);
 struct dpu_extdst *dpu_ed_get(struct dpu_soc *dpu, int id);
 void dpu_ed_put(struct dpu_extdst *ed);
 struct dpu_extdst *dpu_aux_ed_peek(struct dpu_extdst *ed);
-- 
1.7.9.5

